dask/async.py

```diff
@@ -414,18 +421,16 @@ def get_async(apply_async, num_workers, dsk, result, cache=None,
 
     state = start_state_from_dask(dsk, cache=cache)
 
-    tick = [0]
-
     if state['waiting'] and not state['ready']:
         raise ValueError("Found no accessible jobs in dask")
 
+    if not execute_cm:
+        execute_cm = default_cm
+
     def fire_task():
         """ Fire off a task to the thread pool """
-        # Update heartbeat
-        tick[0] += 1
-        # Emit visualization if called for
-        if debug_counts and tick[0] % debug_counts == 0:
-            visualize(dsk, state, filename='dask_%03d' % tick[0])
+        if scheduler_callback:
+            scheduler_callback(dsk, state)
```

Comment:
I can imagine people wanting to use just one of these functions in isolation rather than using both at the same time.  Most basic logging can be done with just `scheduler_callback` in isolation if it also has the key that was just run.

Redundant in the context of both functions but it's a small change that lets us by with one callback in many cases.